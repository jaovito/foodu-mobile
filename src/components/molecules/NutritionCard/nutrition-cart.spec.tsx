import {render} from '@test-utils';
import {mock} from '../../../test/mocks';
import {NutritionCard} from '.';

describe('<NutritionCard /> tests', () => {
  it('should be defined', () => {
    const nutritionCard = render(
      <NutritionCard
        title={mock.nutritionCard.title}
        info={mock.nutritionCard.info}
      />,
    );

    expect(nutritionCard).toBeDefined();
  });

  it('should be able to find the title inside card', async () => {
    const {findByText} = render(
      <NutritionCard
        title={mock.nutritionCard.title}
        info={mock.nutritionCard.info}
      />,
    );

    const text = await findByText(mock.nutritionCard.title);

    expect(text).toBeTruthy();
  });

  it('should be able to find the info inside card', async () => {
    const {findByText} = render(
      <NutritionCard
        title={mock.nutritionCard.title}
        info={mock.nutritionCard.info}
      />,
    );

    const text = await findByText(mock.nutritionCard.info);

    expect(text).toBeTruthy();
  });

  it('should be able to match with the snapshot', () => {
    const nutritionCard = render(
      <NutritionCard
        title={mock.nutritionCard.title}
        info={mock.nutritionCard.info}
      />,
    ).toJSON();

    expect(nutritionCard).toMatchSnapshot();
  });
});
