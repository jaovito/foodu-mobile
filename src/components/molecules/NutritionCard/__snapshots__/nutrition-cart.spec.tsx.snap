// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NutritionCard /> tests should be able to match with the snapshot 1`] = `
<View
  style={
    [
      {
        "borderColor": "#FE8C00",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 1,
        "minWidth": 177,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontWeight="medium"
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_500Medium",
          "fontSize": 31,
        },
        {
          "color": "#FE8C00",
          "fontSize": 27,
          "marginBottom": 16,
        },
      ]
    }
  >
    Energy
  </Text>
  <Text
    fontWeight="medium"
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_500Medium",
          "fontSize": 31,
        },
        {
          "color": "#0D0D0D",
          "fontSize": 31,
        },
      ]
    }
  >
    Kcal
  </Text>
</View>
`;
